src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,206 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size = 0;
+    private  int midIndex = size / 2;
```

Comment:
The `midIndex` variable is redundant and can lead to potential bugs since it's only calculated once when the class is instantiated. It should be calculated each time it's needed within a method.